src/libraries/System.Text.Json/Common/JsonPascalCaseNamingPolicy.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Text.Json
+{
+    internal sealed class JsonPascalCaseNamingPolicy : JsonNamingPolicy
+    {
+        public override string ConvertName(string name)
+        {
+            if (string.IsNullOrEmpty(name) || !char.IsLower(name[0]))
```

Comment:
I'm guessing this just mirrors the behavior of the camel case converter, but shouldn't we also try to have these convert kebab-case and snake_case renderings to PascalCase? What does Json.NET do in that case?

Reply:
cc @YohDeadfall

Reply:
In my honest opinion all converters should behave like kebab and snake case ones, i.e. split an input string into parts, modify them and concatenate. That's exactly how the rest of the world does that, but for historical reasons the camel case policy cannot be changed. Due to that reason I'm thinking that this implementations sounds good, it just mirrors what the snake case does. That's the most expected behavior, I think.

But if it's possible to align the behavior of **both** camel and pascal case policies to what's going on for snake and camel case policies, then better to do so.

Reply:
I was thinking the same thing, it might be something we could turn on with a feature flag in the future.

Reply:
Hey both, sorry for the late replies.

Json.NET doesn't have an implementation for PascalCase exactly, it just returns the name of the property as is in the `DefaultNamingStrategy` - see [here](https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Serialization/DefaultNamingStrategy.cs)

There is an implementation of `CamelCaseNamingStrategy` - [here](https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Serialization/CamelCaseNamingStrategy.cs)
and an implementation of `KebabCaseNamingStrategy` - [here](https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Serialization/KebabCaseNamingStrategy.cs)
These call the StringUtils which also doesn't have an implementation for PascalCase - [here](https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Utilities/StringUtils.cs)

I can look into the current kebab and snake case scenarios and try implement that in this PR for the Pascal Case scenario if that works? I might need some help on how to create feature flags as it would be changing the existing camel case scenarios which I guess would be a breaking change?